### DISCUSSION TITLE
SROS Dial-Out Telemetry: Removing Source Port Information

### DISCUSSION THREAD

**Discussion by ebinachan** [karimra/gnmic#566] [Q&A] [answered]
Hi,

First of all, great work on gNMIc, I've been going over the docs for the last few days and it's impressive!

I've set up dial-out telemetry with SROS, and basic export to InfluxDB works as expected:
```
> select * from PS1
name: PS1
time                           /state/port/transceiver/digital-diagnostic-monitoring/received-optical-power/current port_port-id source         subscription-name
----                           ------------------------------------------------------------------------------------ ------------ ------         -----------------
2022-03-28T08:55:55.010640002Z -3.61                                                                                1/1/1        2.2.2.89:53857 PS1
2022-03-28T08:55:55.040478647Z -40.0                                                                                1/1/3        2.2.2.89:53857 PS1
2022-03-28T08:55:55.08044211Z  -2.02                                                                                1/1/7        2.2.2.89:53857 PS1
```

Next I was trying to set up a processor to extract the source IP address and drop the source port from the DB export. This is where I got stuck, I tried a bunch of things and it seemed like processor section was ignored.
Unfortunately I can't get dial-in to work on this device, and dial-out was my preferred solution anyway.

If anyone has any tips on what I should try next, I would really appreciate it.
[20220329_gnmic_log_1.txt](https://github.com/karimra/gnmic/files/8369741/20220329_gnmic_log_1.txt)

**Comment by karimra** [comment]
Hi, thanks for using gNMIc.
You are not doing anything wrong, the processors are just not loaded with the Nokia dialout telemetry command listen. 
Since it is not coming from the OC gNMI spec, it was intended to be just for demo purposes, not all features are there.

If this is needed I can add it.

BTW, the latest release of SROS supports grpc-tunnel, the OC defined way of doing dialout telemetry.
gNMIc supports grpc-tunnel as well.

**Comment by ebinachan** [comment]
Hi @karimra,

Thank you so much for the answer! Right now we're using classic CLI config across 3000 TiMOS legacy devices, with SNMPv3 Telegraf>InfluxDB monitoring.
Having dial-out with processors enabled would help a lot for newer SROS platforms! 
The initial device config would have a list of sensors enabled, and then the daily cron script would add or update Grafana dashboards.
![20220329_sros_snmp_](https://user-images.githubusercontent.com/52893423/160589339-2766fd26-87ec-4861-ad95-eb4ea9d49fc7.jpg)

**Comment by karimra** [comment]
Sure, I will do that and keep you posted.

**Comment by ebinachan** [comment, marked as answer]
Hi @karimra,

I got 0.24.3 and processors work well with listen mode.
```
processors:
  sourcetrim:
    event-strings:
      tag-names:
        - "source"
      transforms:
        - split:
            apply-on: "value"
            split-on: ":"
            join-with: ""
            ignore-last: 1
```
```
> select * from PS1 where time > now() - 30s
name: PS1
time                           /state/card/cpu/summary/busiest-core-utilization/cpu-usage card_slot-number cpu_sample-period port_port-id source   subscription-name
----                           ---------------------------------------------------------- ---------------- ----------------- ------------ ------   -----------------
2022-04-04T07:48:37.702557818Z 3.79                                                       1                1                              2.2.2.89 PS1
2022-04-04T07:48:37.703563108Z 3.86                                                       1                60                             2.2.2.89 PS1
2022-04-04T07:48:37.70435747Z  3.91                                                       1                300                            2.2.2.89 PS1
```

Thank you so much for all the hard work on this project!
It's really inspiring and pushing me to step up my Go skills.

**Reply by karimra** [reply]
Glad it helps,
You were faster at trying out the new release than me notifying you here :)
